lib/utils/file.js

```diff
@@ -0,0 +1,17 @@
+// Based on https://github.com/kevva/executable
+// Since this module doesn't expose the function to check stat mode only,
+// his logic is pasted here.
+// Opened an issue and a pull request about it,
+// to maybe switch to module in the future:
+// Issue: https://github.com/kevva/executable/issues/9
+// PR: https://github.com/kevva/executable/pull/10
+
+export function isExecutable (fileStat) {
+  if (process.platform === 'win32') return true;
```

Comment:
Consider that it will work not only for drop file onto the dock icon, but also for `open -a '/Applications/Hyperterm.app' 'path/to/a/folder/or/a/file'`. 

So the **folder** feature will be usefull for things like 'atom-terminal' (a plugin for atom that opens the project folder in the terminal) or termHere (an osx application that open the terminal the current finder terminal), since both uses `open -a` under the hood.

Likewise the **executable** may be useful for the same reason, but if you have concerns about that we could reconsider it in the future.

I think that the important thing of this PR is to make file dropping onto the dock icon and `open -a` doing something, because it's a very common and shared behaviour. Enhance this feature could be done on user's feedback basis also in the future :)